ncrypt/smime.c

```diff
@@ -2524,25 +2533,36 @@ int smime_class_send_menu(struct Email *e)
                           _("123")))
               {
                 case 1:
-                  mutt_str_replace(&C_SmimeEncryptWith, "aes128");
+                  rc = cs_subset_str_string_set(NeoMutt->sub,
+                      "smime_encrypt_with", "aes128", &errmsg);
                   break;
                 case 2:
-                  mutt_str_replace(&C_SmimeEncryptWith, "aes192");
+                  rc = cs_subset_str_string_set(NeoMutt->sub,
+                      "smime_encrypt_with", "aes192", &errmsg);
                   break;
                 case 3:
-                  mutt_str_replace(&C_SmimeEncryptWith, "aes256");
+                  rc = cs_subset_str_string_set(NeoMutt->sub,
+                      "smime_encrypt_with", "aes256", &errmsg);
                   break;
               }
               break;
 
             case 4:
-              FREE(&C_SmimeEncryptWith);
+              C_SmimeEncryptWith = NULL;
```

Comment:
Argh!  That's not what I meant.  Sorry, entirely my fault.  I should have been clearer.
I meant to set the string to `NULL` using the config function:
```c
cs_subset_str_string_set(..., NULL, ...);
```

Using the config function will mean old values are freed and notifications are sent.

Reply:
yes sorry about that. I could have just thought about this line a few more seconds.